Framework/PythonInterface/mantid/api/src/Exports/AnalysisDataService.cpp

```diff
@@ -85,5 +85,14 @@ void export_AnalysisDataService() {
       .def("addToGroup", &AnalysisDataServiceImpl::addToGroup, (arg("groupName"), arg("wsName")),
            "Add a workspace in the ADS to a group in the ADS")
       .def("removeFromGroup", &AnalysisDataServiceImpl::removeFromGroup, (arg("groupName"), arg("wsName")),
-           "Remove a workspace from a group in the ADS");
+           "Remove a workspace from a group in the ADS")
+      .def("unique_name", &AnalysisDataServiceImpl::uniqueName,
+           (arg("self"), arg("n") = 5, arg("prefix") = "", arg("suffix") = ""),
+           ""
+           "Return a randomly generated unique name for a workspace \
+           `n` : length of string of random numbers, `prefix` : String to be prepended to the generated string, \
+           `suffix` : String to be appended to the generated string, `returns` : (prefix + n*random characters + suffix) "
+           "")
```

Comment:
```suggestion
      .def("unique_name", &AnalysisDataServiceImpl::uniqueName,
           (arg("self"), arg("n") = 5, arg("prefix") = "", arg("suffix") = ""),
           "Return a randomly generated unique name for a workspace\n"
           "\nParameters\n"
           "----------\n"
           "n: str\n"
           "   length of string of random numbers\n"
           "prefix: str\n"
           "   String to be prepended to the generated string\n"
           "suffix: str\n"
           "   String to be appended to the generated string\n"
           "\nReturns\n"
           "-------\n"
           "str\n"
           "    prefix + n*random characters + suffix\n"
           )
```
This is how it looks:
<img width="592" alt="Capture" src="https://github.com/mantidproject/mantid/assets/1136006/713e377e-13fc-448c-9ab4-dc6b8c0f4154">

Reply:
I tried doing it that way. I'm not sure how you are getting the docstring to print. I was building `docs-html` and checking what it looked like on the `AnalysisDataServiceImpl` page and if you do it the way you suggested it jumbles it all up.